torch/csrc/jit/python/pybind_utils.h

```diff
@@ -163,15 +194,19 @@ struct VISIBILITY_HIDDEN PythonFutureWrapper
                 e.what()));
             {
               pybind11::gil_scoped_acquire ag;
+              py::object exceptionType = e.type();
               // Release ownership on py::objects and also restore Python
               // Error Indicator.
               e.restore();
               // Clear the Python Error Indicator as we has recorded the
               // exception in the response message.
               PyErr_Clear();
+              JitFutureExceptionHandler().handleExceptionGILHeld(
+                  exceptionType, err.what());
             }
-
-            throw err;
+            // python error is already set above, so this is just to satisfy
+            // C++ compiler.
+            return at::IValue();
```

Comment:
can we still throw err here? or because we already raise it in `handleExceptionGILHeld`?